packages/legend-query-builder/src/stores/explorer/QueryFunctionsExplorerState.ts

```diff
@@ -279,18 +307,48 @@ export class QueryFunctionsExplorerState {
     }
   }
 
+  setFunctionInfoMap(info: Map<string, FunctionAnalysisInfo>) {
+    this.functionInfoMap = info;
+  }
+
+  setDependencyFunctionInfoMap(info: Map<string, FunctionAnalysisInfo>) {
+    this.dependencyFunctionInfoMap = info;
+  }
+
+  setPackagePathToFunctionInfoMap(map: Map<string, FunctionAnalysisInfo[]>) {
+    this.packagePathToFunctionInfoMap = map;
+  }
+
   async initializeDisplayablePackagesSet(): Promise<void> {
-    this.queryBuilderState.graphManagerState.graph.ownFunctions
-      .map((f) => getAllPackagesFromElement(f))
-      .flat()
-      .forEach((pkg) => this.displayablePackagesSet.add(pkg));
+    if (this.functionInfoMap) {
+      Array.from(this.functionInfoMap.values())
+        .map((info) =>
+          getOrCreateGraphPackage(
```

Comment:
so we didnt create a new tree to display with the name of the function package. we are changing the graph and adding new packages?

Reply:
I created a simple new graph just for the functions that will be rendered